### ISSUE TITLE
@Suppress with Ruleset name does not work

### ISSUE THREAD

**Issue by eekboom** [detekt/detekt#5976]
The docs at https://detekt.dev/docs/introduction/suppressing-rules/ say that it should be possible to suppress a warning like this @Suppress("complexity.LongParameterList")

So, this should _not_ trigger this warning:

```
@Suppress("complexity.LongParameterList")
private fun foo (a: Int, b: Int, c: Int, d: Int, e: Int, f: Int, g: Int) {
}
```

but it currently does.

detekt 1.22.0 and gradle 8.1-rc-3

**Comment by cortinico** [comment]
Does it work if you just `@Suppress("LongParameterList")`?

**Comment by eekboom** [comment]
Yes, it works without the rule set name.
So it is not that urgent, but I'd like to use the full name so that it is easier to find the rule in the config file.

**Comment by BraisGabin** [comment]
I would say that this is an issue in the documentation instead a bad behavior.

We already check for `"rulename"` and `"detekt.rulename"`. I don't think adding even more options here is a good idea.

**Comment by marschwar** [comment]
And there specifically is a test case for this

```kotlin
@Test
fun `suppresses by combination of rule set and rule id`() {
    assertCodeIsSuppressed("""@Suppress("complexity.LongParameterList")$code""", config)
}
```

I will investigate.

**Comment by marschwar** [comment]
The reason for this behavior is not so obvious. Within the `Rule#report` function we check whether to suppress the finding or not. To determine the name of the ruleset for this purpose, we rely on `Config#parentPath`. The problem is that this is implemented only in the `YamlConfig` and all other Config implementations return null. Therefor the suppression mechanism fails when the suppression is configured with the rule set id.

As the rule - as it stands right now - is unaware of the ruleset it belongs to, the only way to resolve this is to implement the getter correctly in all Config implementations.   Any idea if that will cause any side effects? I will prepare a pr.
